### DISCUSSION TITLE
Does steel scheme support precision math? Can it?

### DISCUSSION THREAD

**Discussion by amano-kenji** [mattwparas/steel#146] [Q&A] [answered]
I imagine that standalone steel scheme programs could be written for finance.

For example, https://hackage.haskell.org/package/Decimal provides precision for up to 255 decimal places. I used it in my financial program. Raku has built-in number types that can handle precision math.

Some helix scripts may also want some financial calculations.... I don't know...

**Comment by mattwparas** [comment]
Steel supports big integers, so I would presume yes it can support precision math

**Reply by amano-kenji** [reply]
Big integers don't come with decimal places......

I was expecting something like https://hackage.haskell.org/package/Decimal or something else that supports more decimal places...

Someone can build something like Decimal on top of big integers, but it doesn't exist, yet.

I would say that arbitrary-precision decimal numbers and limited-precision decimal numbers can be added later.

**Reply by mattwparas** [reply, marked as answer]
Sorry, what I'm saying is that big int support is all that is really necessary to build the support if you really wanted it
